setup.sh

```diff
@@ -7,9 +7,12 @@ echo "Setting environment variable SPACK_STACK_DIR to ${SPACK_STACK_DIR}"
 
 source ${SPACK_STACK_DIR:?}/spack/share/spack/setup-env.sh
 echo "Sourcing spack environment ${SPACK_STACK_DIR}/spack/share/spack/setup-env.sh"
+# Avoid using ~/.spack directory for caching, bootstrap, etc.
 if [ "$(uname)" != "Darwin" ]; then
   export SPACK_USER_CACHE_PATH=$SPACK_ROOT/user_cache
 fi
+export SPACK_DISABLE_LOCAL_CONFIG=1
+echo "Changing bootstrap path to $(spack bootstrap root '$spack/bootstrap')"
```

Comment:
After running the test (https://github.com/JCSDA/spack-stack/pull/1352#issuecomment-2427500838) I see this in my local clone of the spack repo:
```
[test] MacBook-Pro-5:spack steveherbener$ git status
HEAD detached at b6ef265ec4
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	bootstrap/

nothing added to commit but untracked files present (use "git add" to track)
```

This is reflecting the desired outcome, and I think we should add "bootstrap" to the .gitignore file in the spack repo to help avoid accidentally adding the bootstrap directory to a commit.

Reply:
Does this mean that the other issue with the location of the bootstrap dir is resolved?